virttest/qemu_monitor.py

```diff
@@ -2635,20 +2764,23 @@ def set_migrate_parameter(self, parameter, value):
         """
         cmd = "migrate-set-parameters"
         self.verify_supported_cmd(cmd)
+        parameter = self._get_migrate_parameter(parameter, error_on_missing,
+                                                disable_auto_x_evaluation)
         args = {parameter: value}
         return self.cmd(cmd, args)
 
-    def get_migrate_parameter(self, parameter):
+    def get_migrate_parameter(self, parameter, disable_auto_x_evaluation=True):
         """
         Get the value of parameter.
 
         :param parameter: parameter which need to get.
         """
         parameter_info = self.query("migrate-parameters")
+        parameter = self._get_migrate_parameter(parameter,
+                                                disable_auto_x_evaluation=disable_auto_x_evaluation)
```

Comment:
Something like this,
1. could help to address x-.* attributes in a generic manner
2. could avoid calling self.query() multiple times in multiple places

```python
def x_evaluation(self, attribute, error_on_missing=False):
    if attribute.startswith("x-"):
        attribute2 = attribute[2:]
    else:
        attribute2 = "x-%s" % attribute
    if attribute2 in self._supported_attributes:
        return attribute2
    if error_on_missing:
        raise MonitorNotSupportedError("Attribute %s, nor %s "
                                       "supported." % (attribute,
                                                       attribute2))

def query(self, what, attribute=None, disable_auto_x_evaluation=True):
    info = self.info(what)
    if attribute is None:
        return info
    if self._supported_attributes is None:
                self._supported_attributes = info.keys()
    if(disable_auto_x_evaluation or
       attribute in self._supported_attributes):
        return attribute
    return x_evaluation(attribute)


# make it generic for all the query of parameters and capabilities
def get_migrate_parameter(self, parameter, disable_auto_x_evaluation=True):
    self.query("migrate-parameters", attribute="x-multifd-channels"
               disable_auto_x_evaluation=disable_auto_x_evaluation)

def get_migrate_capability(self, capability,  disable_auto_x_evaluation=True):
    self.query("migrate-capabilities", attribute="x-multifd"
               disable_auto_x_evaluation=disable_auto_x_evaluation)
```

Reply:
I see. Given the number of usage I simply hardcoded it, but let's give it a try...